apps/guide/content/docs/legacy/interactions/modals.mdx

```diff
@@ -35,112 +32,330 @@ client.on(Events.InteractionCreate, async (interaction) => {
+			// The label is a large header text that identifies the interactive component for the user.
+			.setLabel('What are some of your favorite hobbies?')
+			// The description is optional small text beneath the label that provides additional information about the interactive component.
+			.setDescription('Activities you like to participate in');
+
+		// [!code ++:2]
+		// Add label to the modal
+		modal.addLabelComponents(hobbiesLabel);
+	}
+});
 ```
 
+<Callout>
+	The `label` field has a max length of 45 characters. The `description` field has a max length of 100 characters.
 </Callout>
 
+### Text input
+
+Text input components prompt users for single or multi line free-form text.
+
 ```js
-const { ActionRowBuilder, Events, ModalBuilder, TextInputBuilder, TextInputStyle } = require('discord.js');
+const { LabelBuilder, ModalBuilder, TextInputBuilder, TextInputStyle } = require('discord.js');
 
 client.on(Events.InteractionCreate, async (interaction) => {
 	if (!interaction.isChatInputCommand()) return;
-
 	if (interaction.commandName === 'ping') {
+		// [!code focus:10]
 		// Create the modal
 		const modal = new ModalBuilder().setCustomId('myModal').setTitle('My Modal');
 
-		// Add components to modal
-
-		// Create the text input components
-		const favoriteColorInput = new TextInputBuilder()
-			.setCustomId('favoriteColorInput')
-			// The label is the prompt the user sees for this input
-			.setLabel("What's your favorite color?")
-			// Short means only a single line of text
-			.setStyle(TextInputStyle.Short);
-
+		// [!code ++:6]
 		const hobbiesInput = new TextInputBuilder()
 			.setCustomId('hobbiesInput')
+			// Short means a single line of text.
+			.setStyle(TextInputStyle.Short)
+			// Placeholder text displayed inside the text input box
+			.setPlaceholder('card games, films, books, etc.');
+
+		// [!code focus:10]
+		const hobbiesLabel = new LabelBuilder()
+			// The label is a large header that identifies the interactive component for the user.
 			.setLabel("What's some of your favorite hobbies?")
-			// Paragraph means multiple lines of text.
-			.setStyle(TextInputStyle.Paragraph);
+			// The description is optional small text beneath the label that provides additional information about the interactive component.
+			.setDescription('Activities you like to participate in')
+			// [!code ++:2]
+			// Set text input as the component of the label
+			.setTextInputComponent(hobbiesInput);
+
+		// Add the label to the modal
+		modal.addLabelComponents(hobbiesLabel);
+	}
+});
+```
+
+#### Input styles
+
+Discord offers two different input styles:
+
+- `Short`, a single-line text entry
+- `Paragraph`, a multi-line text entry
+
+#### Input properties
+
+A text input field can be customized in a number of ways to apply validation or set default values via the following `TextInputBuilder` methods:
+
+```js
+const input = new TextInputBuilder()
+	// Set the component id (this is not the custom id)
+	.setId(0)
```

Comment:
Setting an id of `0` makes Discord treat it as if one did not set an id. Use `1` or higher.

```suggestion
	.setId(1)
```